tap_azure_git/__init__.py

```diff
@@ -333,6 +362,15 @@ def get_catalog():
 
     return {'streams': streams}
 
+
+@functools.cache
+def get_repository(org, project, repo):
```

Comment:
In some places we need the `id` of the repo, but sometimes we only have the `repo_path`.  This will allow us to get the repo and cache the result for future calls with the same params.